internal/security/bootstrapper/command/install/command.go

```diff
@@ -0,0 +1,144 @@
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+)
+
+const (
+	CommandName string = "install"
+)
+
+type cmd struct {
+	cntx          context.Context
+	waitGroup     *sync.WaitGroup
+	loggingClient logger.LoggingClient
+	config        *config.ConfigurationStruct
+}
+
+func NewCommand(
+	ctx context.Context,
+	wg *sync.WaitGroup,
+	lc logger.LoggingClient,
+	conf *config.ConfigurationStruct,
+	args []string) (interfaces.Command, error) {
+
+	cmd := cmd{
+		loggingClient: lc,
+		config:        conf,
+		cntx:          ctx,
+		waitGroup:     wg,
+	}
+	var dummy string
+
+	flagSet := flag.NewFlagSet(CommandName, flag.ContinueOnError)
+	flagSet.StringVar(&dummy, "confdir", "", "") // handled by bootstrap; duplicated here to prevent arg parsing errors
+
+	err := flagSet.Parse(args)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to parse command: %s: %w", strings.Join(args, " "), err)
+	}
+
+	return &cmd, err
+}
+
+func (c *cmd) Execute() (statusCode int, err error) {
+	c.loggingClient.Info(fmt.Sprintf("Security bootstrapper running %s", CommandName))
+
+	bootstrapServer := tcp.NewTcpServer()
+	c.loggingClient.Debug(fmt.Sprintf("init phase: attempts to start up the listener on bootstrap port: %d",
+		c.config.StageGate.BootstrapPort))
+
+	gatingSemaphorePort(bootstrapServer, c.config.StageGate.BootstrapPort, c.loggingClient,
+		"Raised bootstrap semaphore for secure bootstrapping")
+
+	// wait on for others to be done: each of tcp dialers is a blocking call
+	c.loggingClient.Debug("Waiting on dependent semaphores required to raise the ready-to-run semaphore ...")
+	if err := tcp.DialTcp(c.config.StageGate.ConsulHost, c.config.StageGate.ConsulReadyPort, c.loggingClient); err != nil {
+		c.loggingClient.Error(err.Error())
+	}
+	c.loggingClient.Info("Consul is ready")
+
+	if err := tcp.DialTcp(c.config.StageGate.PostgresHost, c.config.StageGate.PostgresReadyPort, c.loggingClient); err != nil {
+		c.loggingClient.Error(err.Error())
+	}
+	c.loggingClient.Info("Postgres is ready")
+
+	if err := tcp.DialTcp(c.config.StageGate.RedisHost, c.config.StageGate.RedisReadyPort, c.loggingClient); err != nil {
+		c.loggingClient.Error(err.Error())
+	}
+	c.loggingClient.Info("Redis is ready")
+
+	// Reached ready-to-run phase
+	c.loggingClient.Debug(fmt.Sprintf("ready-to-run phase: attempts to start up the listener on ready-to-run port: %d",
+		c.config.StageGate.ReadyToRunPort))
+
+	readToRunServer := tcp.NewTcpServer()
+
+	gatingSemaphorePort(readToRunServer, c.config.StageGate.ReadyToRunPort, c.loggingClient,
+		"Raised ready-to-run semaphore for secure bootstrapping")
+
+	// keep running until ctx done
+	c.waitGroup.Add(1)
+	go func() {
+		defer c.waitGroup.Done()
+
+		<-c.cntx.Done()
+		c.loggingClient.Info("security bootstrapper install finished")
+	}()
+
+	return
+}
+
+func (c *cmd) GetCommandName() string {
+	return CommandName
+}
+
+func gatingSemaphorePort(tcpServer *tcp.TcpServer, portNum int, lc logger.LoggingClient, raisedMsg string) {
```

Comment:
How about renaming to `openGatingSemaphorePort`?

Reply:
changed